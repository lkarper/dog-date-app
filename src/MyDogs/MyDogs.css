.MyDogs__list {
    padding: 0;
}

@media screen and (min-width: 750px) {
    .MyDogs__list {
        display: flex;
        flex-direction: row;
        justify-content: space-around;
    }
}