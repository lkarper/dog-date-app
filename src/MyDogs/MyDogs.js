import React, { useContext } from 'react';
import { Link } from 'react-router-dom';
import UserContext from '../contexts/UserContext';
import DogListView from '../DogListView/DogListView';
import './MyDogs.css';

const MyDogs = (props) => {

    const context = useContext(UserContext);
    const { dogs } = context;

    return (
        <section 
            aria-live='polite' 
            className='MyDogs__section section'
        >
            <header>
                <h2>My dogs</h2>
            </header>
            <Link to='/create-dog-profile'>Create a new dog profile!</Link>
            {dogs && (dogs.length > 0
                ?
                    <ul className='MyDogs__list'>
                        {dogs.map(dog => <DogListView key={dog.id} dog={dog} />)}
                    </ul>
                :
                    <p>No dogs currently linked to your account.</p>)
            }
        </section>
    );
}

export default MyDogs;
