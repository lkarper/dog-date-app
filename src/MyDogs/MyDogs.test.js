import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import MyDogs from './MyDogs';

describe('MyDogs component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <MyDogs />
            </BrowserRouter>, 
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(
            <BrowserRouter>
                <MyDogs />
            </BrowserRouter>
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
