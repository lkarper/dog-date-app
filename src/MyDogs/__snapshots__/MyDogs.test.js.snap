// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyDogs component renders the UI as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <MyDogs>
      <section
        aria-live="polite"
        className="MyDogs__section section"
      >
        <header>
          <h2>
            My Dogs
          </h2>
        </header>
        <Link
          className="link"
          to="/create-dog-profile"
        >
          <LinkAnchor
            className="link"
            href="/create-dog-profile"
            navigate={[Function]}
          >
            <a
              className="link"
              href="/create-dog-profile"
              onClick={[Function]}
            >
              Create a new dog profile!
            </a>
          </LinkAnchor>
        </Link>
        <p>
          No dogs currently linked to your account.
        </p>
      </section>
    </MyDogs>
  </Router>
</BrowserRouter>
`;
